daemon/containerd/image_push.go

```diff
@@ -163,14 +166,35 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 	}
 
 	err = remotes.PushContent(ctx, pusher, target, store, limiter, platforms.All, handlerWrapper)
-	if err != nil {
-		if containerdimages.IsIndexType(target.MediaType) && cerrdefs.IsNotFound(err) {
+	if err != nil && containerdimages.IsIndexType(target.MediaType) {
```

Comment:
I don't see the else if case for `err != nil`

Reply:
Thanks, fixed!